### PR TITLE
KAFKA-8736: Streams performance improvement, use isEmpty() rather tha…

### PR DESCRIPTION
**Pull request by mjarvie**
…n size() == 0

According to https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentSkipListMap.html#size--, the size method has to traverse all elements to get a count. It looks like the count is being compared against 0 to determine if the map is empty; In this case, we don't need a full count. Instead, the isEmpty() method should be used, which just looks for one node.

No expected changes to unit or integration tests.

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ableegoldman** [comment]
Thanks for the PR! LGTM. 

I think this should be cherry-picked back to 2.3 cc/ @bbejeck @mjsax @guozhangwang

**Review by mjsax — APPROVED** [review]
LGTM.

Waiting for Jenkins to finish before we can merge.

**Review by guozhangwang — APPROVED** [review]
Thanks @mjarvie ! LGTM.

**Comment by mjsax** [comment]
Java 11 / 2.12 failed with `kafka.server.DescribeLogDirsRequestTest.testDescribeLogDirsRequest`
Java 11 / 2.13 failed with `kafka.server.DescribeLogDirsRequestTest.testDescribeLogDirsRequest`
Java 8 failed with:
```
org.apache.kafka.connect.integration.ExampleConnectIntegrationTest.testSourceConnector
org.apache.kafka.connect.integration.RebalanceSourceConnectorsIntegrationTest.testStartTwoConnectors
kafka.api.SaslSslAdminClientIntegrationTest.testIncrementalAlterConfigsForLog4jLogLevels
kafka.api.SaslSslAdminClientIntegrationTest.testIncrementalAlterConfigsForLog4jLogLevelsCanResetLoggerToCurrentRoot
```

Retest this please

**Comment by mjsax** [comment]
Java 11 / 2.12: `org.apache.kafka.connect.integration.RebalanceSourceConnectorsIntegrationTest.testDeleteConnector`
Java 11 / 2.13: `kafka.server.DescribeLogDirsRequestTest.testDescribeLogDirsRequest`
Java 8: `kafka.server.DescribeLogDirsRequestTest.testDescribeLogDirsRequest`

Merging. Thanks for the PR @mjarvie

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.3`, `2.2`, `2.1`, and `2.0`.

**Pull request merged**
